Deposit table

famole opened the pull request:
Added Sundeep changes.
Fixed table style and created deposit default screen.

famole committed:
First draft of deposit table and overview page

famole committed:
merged sundeeps code

famole committed:
Merged SunDeep changes, Deposits table added, style fixed.

famole committed:
Deposit table removed old implementation

zedkai reviewed (changes requested):
Please change this to no longer use `IMarketDataTable`, the constant `marketData`, or the `BasicTable`.

Instead, please retrieve market data in the same manner as [/src/views/Markets/index.tsx](https://github.com/Agave-DAO/agave-frontend/blob/8d0be47e632abdecb1cf710fc88be96d321684e3/src/views/Markets/index.tsx#L209-L220) and use `SortedHtmlTable` as is done [in that same file](https://github.com/Agave-DAO/agave-frontend/blob/8d0be47e632abdecb1cf710fc88be96d321684e3/src/views/Markets/index.tsx#L308).

famole committed:
Merged dev changes in order to reuse zoe's table, added new props in order to customize header styling

famole committed:
Getting rid of Styled components and built switch using Chakra

famole committed:
Updated Deposits table in order to get data from the correct source and using the generic table data

famole committed:
fixed icons sizes

famole commented:
Fixed data source issue.
Added header props and sort icon to the _sortedHtmlTable_
Switch component updated, styled components implementation was removed.

<img width="1274" alt="Screen Shot 2021-06-09 at 11 27 30 PM" src="https://user-images.githubusercontent.com/1489405/121455429-d617a900-c97a-11eb-9d3c-30d1e4ba7919.png">

famole committed:
useTokenBalance hook added

famole committed:
Added balance values to the table

famole committed:
Fixed token balance inconsisntency

zedkai commented on src/views/Deposit/layout.tsx:
```diff
@@ -118,11 +118,12 @@ export const DepositLayout: React.FC<DepositLayoutProps> = props => {
               />
-              <Search
+              {/* Disabled for now, no enough rows to be filtered */}
+              {/* <Search
```
Added this to a [work-item](https://trello.com/c/20XYvRSR/27-search-functionality-on-deposit-withdrawal-etc-pages-currently-disabled) for later.

zedkai commented on src/hooks/balance.ts:
```diff
@@ -41,3 +41,19 @@ export const useBalance = (asset: IMarketData | undefined): UseBalanceDto => {
   return { balance, address, library, balanceQueryKey };
 };
+
+export const useTokenBalance = (tokenAddress: string): Promise<BigNumber> | undefined => {
```
No, not at all; Use the hooks for this, namely `useUserAssetBalance` which returns an object containing field `data`; when `data` is non-`undefined`, it contains the balance for that user.

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -1,67 +1,162 @@
+    const [ balanceUSD, setBalanceUSD ] = useState('')
+
+    const price = useAssetPriceInDai(tokenAddress);
+    useTokenBalance(tokenAddress)?.then((value: BigNumber) => {
```
Invoking a hook like this is incorrect and should have resulted in React full-on crashing- I'm surprised it didn't.

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -1,67 +1,162 @@
+    const price = useAssetPriceInDai(tokenAddress);
+    useTokenBalance(tokenAddress)?.then((value: BigNumber) => {
+      const result = Number(ethers.utils.formatEther(value) || 0).toFixed(2)
+      setBalance(result)
```
Invoke `useUserAssetBalance` the same way you did with `useAssetPriceInDai` and you would be able to avoid using state at all. Calling `set(State)` from an async callback like this is illegal in React as it can be fulfilled after the component is unmounted, which would trigger a render with no component.

zedkai reviewed (changes requested):
Currently broken; this code would crash if the user selects the page and flips away from it again while on a slow network connection.

famole committed:
removed hook useTokenBalance

famole committed:
Fixed issues identified in PR

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -4,13 +4,14 @@ import { CellProps, Column, Renderer } from "react-table";
 import { TokenIcon } from "../../utils/icons";
 import { BigNumber } from "ethers";
+import { useUserAssetBalance } from "../../queries/userAssets";
+import { FixedFromRay } from "../../utils/fixedPoint";
```
Doesn't look like you use this at all?

famole replied:
Removed all unused imports

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -49,30 +50,26 @@ const DTable: React.FC<{ activeType: string }> = ({ activeType }) => {
-    });
     
+    const price = useAssetPriceInDai(tokenAddress);
+    const balance = useUserAssetBalance(tokenAddress)
```
Semicolons. Use the `prettier` to format.

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -49,30 +50,26 @@ const DTable: React.FC<{ activeType: string }> = ({ activeType }) => {
+    const balance = useUserAssetBalance(tokenAddress)
+    const balanceNumber = balance.data ? Number(ethers.utils.formatEther(balance.data)) : undefined
+    const renderBalanceDai = balanceNumber ? (Number(price.data) * balanceNumber).toFixed(2) : "-"
+    console.log(Number(price.data))
```
Leftover debug log?

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -49,30 +50,26 @@ const DTable: React.FC<{ activeType: string }> = ({ activeType }) => {
+    const price = useAssetPriceInDai(tokenAddress);
+    const balance = useUserAssetBalance(tokenAddress)
+    const balanceNumber = balance.data ? Number(ethers.utils.formatEther(balance.data)) : undefined
+    const renderBalanceDai = balanceNumber ? (Number(price.data) * balanceNumber).toFixed(2) : "-"
```
These look super sketchy, but- since they're only for display- I'll check them for accuracy myself later.

zedkai commented on src/views/Deposit/DepositTable.tsx:
```diff
@@ -49,30 +50,26 @@ const DTable: React.FC<{ activeType: string }> = ({ activeType }) => {
             </Text>
             <Text p={3}>
-              $ {balance ?? "-"}
+              $ {renderBalanceDai ?? "-"}
```
`balance` was originally the balance in tokens, not in Dai. `balanceUsd` would be where the balance in Dai should be placed.

famole replied:
The token balance needs to be on top by design, and the amount in USD under it. Just updated variable naming to make it clear.

famole committed:
Removed unused imports, and fixed small code issues.

zedkai reviewed (approved):
That's better. Merging.
